File: anti-thief-new/app/src/main/java/com/emopass/antitheftalarm/ui/language/LanguageActivity.java
package com.emopass.antitheftalarm.ui.language;

import android.app.Activity;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.WindowManager;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;
import com.ads.control.ads.VioAdmob;
import com.emopass.antitheftalarm.adapter.LanguageAdapter;
import com.emopass.antitheftalarm.databinding.ActivityLanguageBinding;
import com.emopass.antitheftalarm.model.LanguageModel;
import com.emopass.antitheftalarm.ui.BaseActivity;
import com.emopass.antitheftalarm.ui.onboarding.OnBoardingActivity;
import com.emopass.antitheftalarm.utils.AdsUtils;
import com.emopass.antitheftalarm.utils.LanguageUtils;
import com.ironsource.mediationsdk.IronSource;

import java.util.List;

public class LanguageActivity extends BaseActivity<ActivityLanguageBinding> {
    LanguageAdapter languageAdapter;
    private LanguageViewModel viewModel = null;
    private Boolean isFromSetting = false;
    @Override
    protected void initView() {
        LanguageUtils.getInstance().loadLocale(this);
        AdsUtils.getInstance().requestNativeLanguage(this, false);
        isFromSetting = getIntent().getBooleanExtra(FROM_SETTING_KEY, false);
        viewModel = new ViewModelProvider(this).get(LanguageViewModel.class);
        setUpAdapter();

    }

    @Override
    protected ActivityLanguageBinding getViewBinding() {
        return ActivityLanguageBinding.inflate(LayoutInflater.from(this));
    }

    @Override
    protected void initControl() {
        binding.imgChoose.setOnClickListener(v -> {
            LanguageUtils.getInstance().changeLang(languageAdapter.itemSelected().getLanguageCode(), this);
            if(!isFromSetting) {
                launchOnBoard();
            }
        });
        binding.imgBack.setOnClickListener(v -> {
            onBackPressed();
        });
        viewModel.selectedLanguage.observe(this, new Observer<String>() {
            @Override
            public void onChanged(String s) {
                int visible;
                if(s.isEmpty() || s == null) {
                    visible = View.GONE;
                }else {
                    visible = View.VISIBLE;
                }
                binding.imgChoose.setVisibility(visible);
            }
        });
        if(isFromSetting) {
            binding.frAds.setVisibility(View.GONE);
            binding.imgBack.setVisibility(View.VISIBLE);
        }else {
            binding.imgBack.setVisibility(View.GONE);
            initAds();
            hideNavigationBar();
        }
    }

    @Override
    public void onWindowFocusChanged(boolean hasFocus) {
        super.onWindowFocusChanged(hasFocus);
        if(hasFocus && !isFromSetting) {
            hideNavigationBar();
        }
    }

    @Override
    protected void onResume() {
        super.onResume();
        IronSource.onResume(this);
    }

    @Override
    protected void onPause() {
        super.onPause();
        IronSource.onPause(this);
    }

    private void setUpAdapter() {
        languageAdapter = new LanguageAdapter(this);
        languageAdapter.setOnLanguageChange(new LanguageAdapter.OnLanguageChange() {
            @Override
            public void onChange(String language) {
                viewModel.selectLanguage(language);
            }
        });
        List<LanguageModel> listLang = LanguageUtils.getInstance().getSupportedLanguages();
        if(!isFromSetting) {
            listLang = listLang.subList(0,5);
        }
        languageAdapter.setData( listLang);
        LinearLayoutManager layoutManager = new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false);
        binding.rvcLanguage.addItemDecoration(new SpaceDecorator(8));
        binding.rvcLanguage.setLayoutManager(layoutManager);
        binding.rvcLanguage.setAdapter(languageAdapter);
    }

    private void hideNavigationBar() {
        int layoutInDisplayCutoutModeDefault = WindowManager.LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_DEFAULT;
        int flags = (View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
        | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
        | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
        | View.SYSTEM_UI_FLAG_FULLSCREEN
        | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);

        getWindow().getDecorView().setSystemUiVisibility(flags);
        getWindow().getDecorView().setOnSystemUiVisibilityChangeListener(new View.OnSystemUiVisibilityChangeListener() {
            @Override
            public void onSystemUiVisibilityChange(int visibility) {
                if((visibility & View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION) == 0) {
                    getWindow().getDecorView().setSystemUiVisibility(flags);
                }
            }
        });
    }

    private void launchOnBoard() {
        Intent intent = new Intent(this, OnBoardingActivity.class);
        startActivity(intent);
        finish();
    }

    private void initAds() {
        AdsUtils.nativeLanguageLoadFail.observe(this, new Observer<Boolean>() {
            @Override
            public void onChanged(Boolean aBoolean) {
                if(aBoolean) {
                    binding.frAds.setVisibility(View.GONE);
                }
            }
        });

        AdsUtils.nativeLanguage.observe(this, apNativeAd -> {
            if(apNativeAd != null) {
                VioAdmob.getInstance().populateNativeAdView(
                        LanguageActivity.this,
                        apNativeAd,
                        binding.frAds,
                        binding.includeNative.shimmerContainerBanner);
            }
        });
    }
    private static String FROM_SETTING_KEY = "from setting";
    public static void start(Activity activity, Boolean fromSetting) {
        Intent intent = new Intent(activity, LanguageActivity.class);
        intent.putExtra(FROM_SETTING_KEY, fromSetting);
        activity.startActivity(intent);
        if(!fromSetting) {
            activity.finish();
        }
    }
}

File: findmyphone/app/src/main/java/com/example/claptofindphone/presenter/language/LanguageActivity.kt
package com.example.claptofindphone.presenter.language

import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.view.ViewGroup
import android.view.WindowManager
import androidx.activity.viewModels
import androidx.core.view.isVisible
import androidx.lifecycle.distinctUntilChanged
import androidx.recyclerview.widget.LinearLayoutManager
import com.ads.control.ads.VioAdmob
import com.example.claptofindphone.data.remote.RemoteConfigProvider
import com.example.claptofindphone.databinding.ActivityLanguageBinding
import com.example.claptofindphone.presenter.MainActivity
import com.example.claptofindphone.presenter.onboarding.OnBoardingActivity
import com.example.claptofindphone.utils.AdsHelper
import com.example.claptofindphone.utils.LanguageSupporter
import com.ironsource.mediationsdk.IronSource
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint
class LanguageActivity : AppCompatActivity() {

    private lateinit var layoutBinding: ActivityLanguageBinding
    private lateinit var languageListAdapter: LanguageAdapter
    private var isFromSetting = false

    @Inject
    lateinit var languageSupporter: LanguageSupporter
    @Inject
    lateinit var adsHelper: AdsHelper
    @Inject
    lateinit var remoteConfigProvider: RemoteConfigProvider
    private val viewModel by viewModels<LanguageScreenViewModel>()
    companion object {
        private const val FROM_SETTING_KEY = "from_setting"
        fun start(activity: Activity, fromSetting: Boolean = false) {
            Intent(activity, LanguageActivity::class.java).apply {
                putExtra(FROM_SETTING_KEY, fromSetting)
                activity.startActivity(this)
                if(!fromSetting) {
                    activity.finish()
                }
            }
        }
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        languageSupporter.loadLocale(this)
        layoutBinding = ActivityLanguageBinding.inflate(layoutInflater)
        setContentView(layoutBinding.root)
        //layoutBinding.frAds.isVisible = adsHelper.isAdsEnabled
        isFromSetting = intent.getBooleanExtra(FROM_SETTING_KEY, false)
        if (isFromSetting) {
            layoutBinding.frAds.visibility = View.GONE
            layoutBinding.imgBack.visibility = View.VISIBLE
            layoutBinding.bottomLayout.visibility = View.GONE
        } else {
            layoutBinding.imgBack.visibility = View.GONE
            val isNativeOnboardingCtaTop = remoteConfigProvider.isNativeAdOnboardCtaTop
            adsHelper.loadNativeOnBroad(this, ctaTop =  isNativeOnboardingCtaTop)
            adsHelper.loadBannerOnboard()
            initAds()
            initBanner()
            //hideNavigationBar()
        }
        setupAdapter()
        layoutBinding.imgChoose.setOnClickListener {
            languageSupporter.changeLang(languageListAdapter.itemSelected().languageCode, this)
            if (isFromSetting) {
                restartApp()
            } else {
                launchOnBoarding()
            }
        }
        layoutBinding.imgBack.setOnClickListener {
            onBackPressed()
        }

        viewModel.selectedLanguageState.observe(this) {
            layoutBinding.imgChoose.isVisible = !it.isNullOrEmpty()
        }
    }

    public override fun onResume() {
        super.onResume()
        IronSource.onResume(this)
    }

    public override fun onPause() {
        super.onPause()
        IronSource.onPause(this)
    }

    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if(hasFocus && !isFromSetting) {
            //hideNavigationBar()
        }
    }

    private fun restartApp() {
        val startAppIntent = Intent(this, MainActivity::class.java).apply {
            flags = Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP
            putExtra("play-splash", false)
        }
        startActivity(startAppIntent)
    }

    private fun initBanner() {
        adsHelper.bannerAdViewLangauge.distinctUntilChanged().observe(this) {
            it?.let {
                (it.parent as? ViewGroup)?.removeView(it)
                it.layoutParams = ViewGroup.LayoutParams(
                    ViewGroup.LayoutParams.MATCH_PARENT,
                    ViewGroup.LayoutParams.WRAP_CONTENT
                )

                layoutBinding.bottomLayout.addView(it)
            }
        }
    }
    private fun initAds() {
        return
        adsHelper.languageApNativeAdLoadFail.observe(this) {
            if (it) {
                layoutBinding.frAds.visibility = View.GONE
                adsHelper.languageApNativeAdLoadFail.value = false
            }
        }
        adsHelper.languageApNativeAd.observe(this) {
            Log.d("AdsLanguage", "${it}")
            if (it != null) {
                VioAdmob.getInstance().populateNativeAdView(
                    this,
                    it,
                    layoutBinding.frAds,
                    layoutBinding.includeNative.shimmerContainerBanner
                )
            }
        }
    }

    private fun setupAdapter() {
        languageListAdapter = LanguageAdapter(this)
        languageListAdapter.setOnLanguageChange {
            viewModel.selectLanguage(it)
        }
        languageListAdapter.setData(
            if (isFromSetting)
                languageSupporter.getSupportedLanguage()
            else
                languageSupporter.getSupportedLanguage().take(5)
        )
        val layoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
        layoutBinding.rvcLanguage.addItemDecoration(SpacingDecorator(8))
        layoutBinding.rvcLanguage.layoutManager = layoutManager
        layoutBinding.rvcLanguage.adapter = languageListAdapter
    }

    private fun launchOnBoarding() {
        Intent(this, OnBoardingActivity::class.java).apply {
            startActivity(this)
            finish()
        }
    }

    fun hideNavigationBar() {
        WindowManager.LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_DEFAULT
        val flags = (View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
                or View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
                or View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                or View.SYSTEM_UI_FLAG_FULLSCREEN
                or View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY)
        window.decorView.systemUiVisibility = flags
        window.decorView.setOnSystemUiVisibilityChangeListener {
            if ((it and View.SYSTEM_UI_FLAG_FULLSCREEN) == 0) {
                window.decorView.systemUiVisibility = flags
            }
        }
    }
}